### PR TITLE
Docker + Singularity + Conda Update 

### PR DESCRIPTION
**Pull request by apeltzer**
This adds a Docker and Singularity badge to smrnaseq. 
Will also work on conda recipe now... 

- Docker Badge
- Singularity Badge

### CODE REVIEW

**Comment by apeltzer** [comment]
Hi @ewels , can you have a look at the missing NGI-visualizations part in the `environment.yml`?  Not quite sure whether this is still required and/or could be replaced by custom content in multiqc by now. 

Everything else should be in the conda file now! Dockerfile updated too, adding a Singularity file now too :-)

**Comment by ewels** [comment]
Super nice work! Thanks @apeltzer - you nicely picked this up where I left off 😀 

The script used in the NGI visualizations package is this one: https://github.com/NationalGenomicsInfrastructure/ngi_visualizations/tree/master/ngi_visualizations/biotypes

We use featureCounts directly in the RNA pipeline to count biotype overlaps now (the first plot), but I'm not aware of any other tools kicking around which plot this as a function of read length - this is the interesting one for this pipeline. I wrote this script years ago now though, so it's possible that something has been released since then.

Suggestion: we should be able to use pip to install this package from the conda environment file, right? That could be a short term fix. In the long term we could try to find a more official tool to do this.

I've been putting this off a little because I'd really like to use a new QC tool from a group at SciLifeLab called miRTrace: https://friedlanderlab.org/research/ - we've been testing this internally and it's great, but they haven't publicly released it yet. I'll give the PI another nudge and ask if there's any progress on this front.

Phil

File: environment.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  - bioconductor-edger=3.20.7
+  - bioconductor-limma=3.28.21
+  - conda-forge::r-statmod=1.4.30
+  - conda-forge::r-data.table=1.11.4
+  - conda-forge::r-gplots=3.0.1
+  - conda-forge::r-r.methodss3=1.7.1
+  - htseq=0.9.1
+  - fastx_toolkit=0.0.14
+
+#Missing NGI-visualizations? Do we need this or is this something that can go?
```
Comment (ewels) [inline environment.yml]: I _think_ that this should work..?

```yaml
  # Install NGI Visualizations
  # TODO: Replace dependency / conda-ise somehow
  - pip:
     - "--editable=git+https://github.com/NationalGenomicsInfrastructure/ngi_visualizations.git"
```

File: environment.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# You can use this file to create a conda environment for this pipeline:
+#   conda env create -f environment.yml
+name: nfcore-smrnaseq-1.0
+channels:
+  - bioconda
+  - conda-forge
+  - defaults
+dependencies:
+  # bismark
```
Comment (ewels) [inline environment.yml]: Can remove this comment..

**Pull request merged**